### PR TITLE
fix: [CNI] handle getting endpoints when state file is empty

### PR DESCRIPTION
**Pull request by matmerr**
**Reason for Change**:

When reconciling with CNI, it may be the case that CNI state hasn't been created yet. Don't block consumers of that state when empty.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

**Review by neaggarwMS — DISMISSED** [review]
:shipit:

File: store/json.go [hunk 1]
```diff
@@ -71,8 +71,18 @@ func (kvs *jsonFileStore) Read(key string, value interface{}) error {
...
 		defer file.Close()
 
+		b, err := ioutil.ReadAll(file)
+		if err != nil {
+			return err
+		}
+
+		if len(b) == 0 {
+			log.Printf("Unable to read file %s, was empty", kvs.fileName)
+			return nil
```
Comment (tamilmani1989) [inline store/json.go:81]: i would suggest return ErrEmptyFile and handle this in restore function (network/manager.go and ipam/manager.go) as like ErrKeyNotFound

File: cni/client/client_integration_test.go [hunk 2]
```diff
@@ -59,8 +59,7 @@ func TestMain(m *testing.M) {
 
 // todo: enable this test in CI, requires built azure vnet
 func TestGetStateFromAzureCNI(t *testing.T) {
-
-	c := AzureCNIClient{exec: exec.New()}
+	c := New(exec.New())
```
Comment (tamilmani1989) [inline cni/client/client_integration_test.go:62]: add a test case with empty file and check it returns appropriate error

**Review by tamilmani1989 — CHANGES_REQUESTED** [review]

**Review by tamilmani1989 — DISMISSED** [review]
lgtm

File: store/json.go (hunk 1 above)
Comment (neaggarwMS) [reply store/json.go]: @tamilmani1989 , this store file is also read by CNS to save CNS state, if are returning error then we need to handle it in there as well.

**Review by neaggarwMS — APPROVED** [review]
:shipit:
